src/cache/cache_operations.rs

```diff
@@ -0,0 +1,151 @@
+use crate::cache::cache::Cache;
+use crate::cache::cache_op_executors::{cache_response, request_cached};
+use crate::config::{request_cache_duration, request_error_cache_duration};
+use crate::utils::errors::ApiResult;
+use rocket::response::content;
+use serde::Serialize;
+
+pub enum Database {
+    Info = 1,
+    Default = 2,
+}
+
+pub struct Invalidate {
+    pattern: String,
+    database: Database,
+}
+
+impl Invalidate {
+    pub fn new() -> Self {
+        Invalidate {
+            pattern: String::new(),
+            database: Database::Default,
+        }
+    }
+
+    pub fn pattern(&mut self, pattern: String) -> &mut Self {
+        self.pattern = pattern;
+        self
+    }
+
+    fn database(&mut self, database: Database) -> &mut Self {
+        self.database = database;
+        self
+    }
+
+    pub fn execute(&self, cache: &impl Cache) {
+        cache.invalidate_pattern(&self.pattern)
+    }
+}
+
+pub struct CacheResponse<'a, R>
+where
+    R: Serialize,
+{
+    database: Database,
+    pub key: String,
+    pub duration: usize,
+    pub resp_generator: Box<dyn Fn() -> ApiResult<R> + 'a>,
+}
+
+impl<'a, R> CacheResponse<'a, R>
+where
+    R: Serialize,
+{
+    pub fn new() -> Self {
+        CacheResponse {
+            database: Database::Default,
```

Comment:
I am a big hesitant about these defaults. I would probably work with `Option` and have them as default set to `None` and enforce on `execute` that all the required params are `Some`

Reply:
They save up a lot of lines of code, I would say. I see your point though, but also in the case of the database `Database::Info` is for a very small subset of requests coming from a specific source file. I personally find it ok in this case.

Reply:
but this will cause that you forget to set it and use the wrong one. While for the database at least right now we can obviously make it a default (as no other one is used) I would not set defaults for the other params (e.g. `key`, `duration` and `resp_generator`).

Reply:
As mentioned in our call on friday, for me the perfect builder would not compile if the required params are not set.

Reply:
Agreed, I am not sure if I can do a compile time check yet. I remember agreeing on forking into issues, that i linked in the description. #343 

I will restore the original behaviour, which was setting the timeout to 0 and having a check in the method that would use the internal default of reqwest, as I thought that defining this values would be the focus of #340